### PR TITLE
Rename macos-builder.nix -> nix-builder-vm.nix

### PR DESCRIPTION
**Pull request by roberth**
## Motivation

This is a fairly general module, and it would be good to use it for other purposes. This will not happen when everything signals that it's for macOS only.

- It should be able to run on Linux (e.g. locked down non-NixOS Linux systems that allow virtualization), or NixBSD.
- It could be generalized to support CPU emulation (if it doesn't already), which is useful for testing cross-built outputs in an accurate environment, as process-level emulation is quirky)

It's not a full rebranding of the module and package, because not-100%-identical renames get bad diff rendering. More can be done in follow-up PRs.

## Things done

Rename the macos builder / darwin builder / linux builder, because all these names are too specific.

`nix-builder-vm.nix` covers its role in NixOS. This file could be ported to a project like NixBSD using the same file name.

## Things to do later

NixOS itself is Linux-specific, so the package name `linux-builder` is ok. I don't think it has to be in the `darwin` package set necessarily.

The package / module should get its own test; which is perfectly feasible as we're already running NixOS VM tests on darwin as the VM host; this test would similarly run qemu in the darwin Nix sandbox.

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Enzime — APPROVED** [review]
LGTM

I've been wanting to make this change for a while. I'm fairly sure cross-arch support should work, so I think it should just work for running an `aarch64-linux` builder on `x86_64-linux` without much extra code

File: nixos/modules/profiles/nix-builder-vm.nix [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+      getty.autologinUser = user;
+
+      openssh = {
+        enable = true;
+
+        authorizedKeysFiles = [ "${keysDirectory}/%u_${keyType}.pub" ];
+      };
+    };
+
+    system.build.macos-builder-installer =
```
Comment (Enzime) [inline nixos/modules/profiles/nix-builder-vm.nix:149]: Not sure if we want to change this attribute as well

Comment (roberth) [reply nixos/modules/profiles/nix-builder-vm.nix:149]: I think so, but let's do the rest later, so that we get a reasonable diff.

Comment (Enzime) [reply nixos/modules/profiles/nix-builder-vm.nix:149]: Could still be done in a separate commit in this PR that keeps the first commit as the specific rename commit

**Review by Gabriella439 — APPROVED** [review]

**Pull request merged**